src/expr/proof_rule.h

```diff
@@ -674,6 +674,49 @@ enum class PfRule : uint32_t
   // ---------------------
   // Conclusion: F
   ARRAYS_TRUST,
+  
+  //================================================= Datatype rules
+  // ======== Unification
+  // Children: (P:(= (C t1 ... tn) (C s1 ... sn)))
+  // Arguments: (i)
+  // ----------------------------------------
+  // Conclusion: (= ti si)
```

Comment:
Could you add a bit of general information here? I guess it mostly boils done to a very basic explanation of what DT is all about. I guess `C` stands for "Constructor"?